smokey/src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -164,7 +169,14 @@ public DrivetrainSubsystem(SubsystemCollection collection) {
    * @param updatedChassisSpeeds - the updated chassis speeds (x, y and rotation)
    */
   public void drive(ChassisSpeeds updatedChassisSpeeds) {
```

Comment:
I am not sure we will get correct movement if we apply the speedReductionFactor to the ChassisSpeeds.  I think we will get more expected behavior if we apply it to the module speeds.  In the same way, we used Constants.DriveVoltageScalar to modulate MAX_VOLTAGE on line 47.  So, I think we should just keep track of the scale factor and multiply it in on lines 278-281.  
If we are going to introduce this scale factor, I think we can remove the Constants.DriveVoltageScalar altogether.

Reply:
agree ... making change